### ISSUE TITLE
Medical - Make the "Shared Equipment" setting local

### ISSUE THREAD

**Issue by Zman6258** [acemod/ACE3#8085]
**Is your enhancement related to a problem?**
There's recently been some discussion within the unit I play in over which of the two "shared equipment" settings to use, Patient First or Medic First. Some are of the opinion that medics shouldn't use a normal troop's supplies while treating them and risk them having low supplies when going to their next engagement, others think that the medic's supplies should be the reserve equipment that's pulled from by infantry who need it, and infantry's supplies should be used first.

**Solution you'd like:**
Pretty much what the title says. Allow each medic to select which setting to use according to their personal preference; if a mission maker or server owner wants to enforce a specific setting, CBA already offers the ability to override clients.

I'm not sure if there's some technical restriction preventing this from being the case, but if possible, I think it'd be an improvement over having it dictated by the mission/server.

**Comment by BaerMitUmlaut** [comment]
Who would have priority if the patient has a different setting than the medic?
IMO, this just overcomplicates things, both for players and for the implementation. Is it really hard for you to make a decision here?

**Comment by severgun** [comment]
> Who would have priority if the patient has a different setting than the medic?
The one who heals. Isn't it obvious?
You heal, you decide.

**Comment by BaerMitUmlaut** [comment]
I don't think this is obvious. You argue either way.

**Comment by Italiandrache** [comment]
> I don't think this is obvious. You can argue either way.

Why not a serverside setting to decide whom clientside settings should have priority?

**Comment by BaerMitUmlaut** [comment]
Now it's getting ridiculous. Want a per-item-setting as well?

**Comment by Zman6258** [comment]
> The one who heals. 

That's pretty much the idea; if you're doing the healing, you decide which supplies to use first, as you'd be able to do IRL.

**Comment by kymckay** [comment]
I understand the argument being made here, but don't think it's the right argument to make.

The consequence of this is that anyone whose setting is to use the patients supplies is being given precedence over anyone else's preference that they didn't. It also introduces behavioural inconsistency (if X heals you they'll use your bandages, if Y heals you they won't - so depending on your preference you want X/Y to heal you). This is really a playstyle convenience setting which should be agreed amongst a community or on a per-scenario basis.

The argument that might be valid is whether this should be done since communities can globally force a setting regardless - so it allows communities to choose if they want to enable their players to prioritise their own preferences instead of coming to an agreement. However, the downside to this is that it would become the default behaviour for this setting which I don't think fits with our intended use case.

**Comment by Zman6258** [comment]
> The consequence of this is that anyone whose setting is to use the patients supplies is being given precedence over anyone else's preference that they didn't. It also introduces behavioural inconsistency (if X heals you they'll use your bandages, if Y heals you they won't - so depending on your preference you want X/Y to heal you). This is really a playstyle convenience setting which should be agreed amongst a community or on a per-scenario basis.

I don't think it's fair to prescribe what communities should or shouldn't do, that's the point of having addon options. Some, and probably most, communities will agree on one or the other and enforce the option, but some might rather leave the setting up to their medics. The behavioral inconsistencies aren't something I'd consider a concern for the same reason; any server or mission that's set up to enforce one of two behaviors will see no change, whereas players in communities who leave such a thing open to change will likely be accepting of that inconsistency.

> The argument that might be valid is whether this should be done since communities can globally force a setting regardless - so it allows communities to choose if they want to enable their players to prioritise their own preferences instead of coming to an agreement. However, the downside to this is that it would become the default behaviour for this setting which I don't think fits with our intended use case.

That was the intent, that communities who wished to force it one way or another would be able to set it as such. I'm also not sure that such a thing _should_ be seen as an "intended use case", personally; I'm in favor of greater customization when none of the functionality of forcing settings is removed. If it really is such a huge issue, an easy alternate solution would be a third option that's set to "Use Player Preference" or something, with another clientside option used only with that setting... but again, that feels redundant considering the exact same behavior can already be accomplished by just allowing it to be forced one way or the other.

In summary, it's an additional option for those who want it, and can be forced for those who prefer its current behavior, and I would always prefer that people be given the option instead of options being forced upon them.

**Comment by commy2** [comment]
"clientside" "serverside"
Am I on the BI forums? lmao

**Comment by Zman6258** [comment]
> "clientside" "serverside"
> Am I on the BI forums? lmao

Yes, not everybody who has suggestions is somebody who's scripted for Arma extensively. I used those terms as the closest analogues I could think of for whatever the "real" term is, and as long as you can understand the intent, who cares?

**Comment by BaerMitUmlaut** [comment]
> if you're doing the healing, you decide which supplies to use first, as you'd be able to do IRL.

I still disagree with this. As a medic, you probably want to use the patients supplies. If you run out, you'd have to go collect material from other non-medics, which is not only annoying, but also time consuming and might result in the loss of a player. If the players run out of bandages, you can always resupply them or they can take something from your backpack. Why would you ever want to use up your own supplies first, what's the advantage here for you? Why would you ever set it to something else?

---

Oh and if you want to argue what happens IRL: This depends on combat doctrine and is not decided by the patient or the medic. For example, in the US military the patient's IFAK is used first. What do you think, they have a nice little discussion first about who's more important? So if you want "realism", the setting should stay as it is.

**Comment by kymckay** [comment]
> any server or mission that's set up to enforce one of two behaviors will see no change

Which makes this a disruptive change, since currently no server or mission is set up this way.

>I'm also not sure that such a thing should be seen as an "intended use case"

Every setting has an intended use inherently. The proposed change significantly changes the intended use of this particular setting, which seems relevant to consider.

>I would always prefer that people be given the option instead of options being forced upon them

I tend to agree, but UX design is not always this simple (which is the crux of this conversation, it's not about function it's about use case and design).

I think this is a worthwhile discussion, but also don't think it's so simple and obvious as you make out. Also keep in mind this proposal complicates the code and UX for (what I would call) an edge case usage (as you say, "probably most" would want to force this setting).

**Comment by Zman6258** [comment]
> I think this is a worthwhile discussion, but also don't think it's so simple and obvious as you make out. Also keep in mind this proposal complicates the code and UX for (what I would call) an edge case usage (as you say, "probably most" would want to force this setting).

See, I appreciate this as a counterargument, instead of "this is bad"; I assumed it would be a fairly simple change, but the user experience hit isn't something I had considered, particularly in regards to servers that have had their settings in place for quite a while. I'm still obviously in favor of it and think the extra flexibility would be worth it, but I can see why it might not be so simple.

**Comment by BaerMitUmlaut** [comment]
No one dismissed this by just calling it bad. I'm not sure how you got that impression.

**Comment by Drofseh** [comment]
I don't understand why this is controversial at all, if i am the person doing the bandaging I should be the person who decides which bandage gets used. If a community is so silly as to force a specific country's doctrine for bandage priority then that community should be able to force force the setting.

**Comment by LupusTheCanine** [comment]
I think that this setting should be able to be chosen by a player as it is reasonable that a medics will prefer to use their own equipment as they typically carry a lot of medical supplies while a soldier should use buddy's medicaments to heal him to not risk running out.

**Comment by Drofseh** [comment]
I still think this would be a good change
